packages/wdio-browserstack-service/src/util.ts

```diff
@@ -343,11 +343,20 @@ export const validateCapsWithA11y = (deviceName?: any, platformMeta?: { [key: st
     return false
 }
 
-export const shouldScanTestForAccessibility = (suiteTitle: string | undefined, testTitle: string, accessibilityOptions?: { [key: string]: any; }) => {
+export const shouldScanTestForAccessibility = (suiteTitle: string | undefined, testTitle: string, accessibilityOptions?: { [key: string]: any; }, world?: { [key: string]: any; }, isCucumber?: boolean ) => {
     try {
         const includeTags = Array.isArray(accessibilityOptions?.includeTagsInTestingScope) ? accessibilityOptions?.includeTagsInTestingScope : []
         const excludeTags = Array.isArray(accessibilityOptions?.excludeTagsInTestingScope) ? accessibilityOptions?.excludeTagsInTestingScope : []
 
+        if (isCucumber) {
+            const tagsList: string[] = []
+            world?.pickle?.tags.map((tag: { [key: string]: any; }) => tagsList.push(tag.name))
+            const excluded = excludeTags?.some((exclude) => tagsList.includes(exclude))
+            const included = includeTags?.length === 0 || includeTags?.some((include) => tagsList.includes(include))
```

Comment:
Could you explain why we do an explicit check here `includeTags?.length === 0`? I feel like the include and exclude arrays should be treated the same way so it's a surprise to see this.

Reply:
Use case: If the user has not specified any includeTags, we should return the shouldScanTestForAccessibility results as true. However, if the user has mentioned a value in includeTags, we should evaluate against it. If the user does not provide excludeTags, no tags should be excluded by default. We will only honour it if the user explicitly specifies tags in excludeTags.